### PR TITLE
MCS code

### PR DESCRIPTION
**Pull request by mmartinoli87**
Uploading MCS R code

### CODE REVIEW

File: black_it/MCS.R [hunk 1]
```diff
@@ -0,0 +1,105 @@
+# Reading the data
+
+mData <- read.csv("Data.csv")
```
Comment (AldoGl) [inline black_it/MCS.R:3]: What is the content of the mData matrix? Specifically, I see it has 4 columns but what do those columns represent exactly?

File: black_it/MCS.R [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+attach(mData)
+
+fnComplete <- function(x,l=301) {
```
Comment (AldoGl) [inline black_it/MCS.R:6]: What does this function do? A line of comment would be useful here

File: black_it/MCS.R [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+	return(c(x,rep(x[length(x)],l-length(x))))
+}
+
+# Building the MCS
+
+vZ <- 300+4.9*(1:301)
```
Comment (AldoGl) [inline black_it/MCS.R:12]: What is vZ?

File: black_it/MCS.R [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+mDist <- NULL
+for (i in 1:16) {
```
Comment (AldoGl) [inline black_it/MCS.R:15]: What does this for loop do? and the following one? Just two likes of comments could be really helpful

File: black_it/MCS.R [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+	mDist	<- rbind(mDist,mDist2)
+	mDataCOP <- NULL
+}
+mDist <- as.data.frame(mDist)
+names(mDist) <- c("Dist","COP")
+
+library(data.table) 
+mD <- setDT(mDist)[,list(mean=mean(Dist),var=var(Dist)),by=c("COP")]
+
+fnTest <- function(vMean,vVar,iN,dA=1) {
```
Comment (AldoGl) [inline black_it/MCS.R:34]: Comments here would also be useful

File: black_it/MCS.R [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+		} else {
+		dQuan <- 0
+		dPVa <- 1
+	}
+	dResult <- 1*(dTest>dQuan)
+	if (is.na(dResult)) dResult <- 0
+	return(list(test=dTest,q=dQuan,p=dPVa,result=dResult))
+}
+
+fnElim	<- function(vMean,vVar,vIndex) {
```
Comment (AldoGl) [inline black_it/MCS.R:56]: Same here

File: black_it/MCS.R [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+	iM <-	which.max(vMean)
+	return(list(mean=vMean[-iM],var=vVar[-iM],indices=vIndex[-iM],elim=vIndex[iM]))
+}
+
+fnMCS <- function(vMean,vVar,vIndex,dA,verbose=1) {
```
Comment (AldoGl) [inline black_it/MCS.R:61]: Same here

**Review by AldoGl — COMMENTED** [review]
Thanks @mmartinoli87 for your code. In order to move forward it would be useful if you could add some comments, particularly -but not exclusively- explaining what the various functions do even in abstract terms, and perhaps  also describing the meaning of the key variables (see e.g., the questions I made on the code in my review).

These comments would be very useful in the transition of the code from R to Python, because it would allow a Python programmer to easily interpret the code without being an expert in R. Moreover, we can then add your comments in the docstrings of the new Python code.

You could add your code comments directly within the "MCS.R" module. Alternatively you can write them here in the chat and I can update the file.

File: black_it/MCS.R (hunk 7 above)
Comment (mmartinoli87) [reply black_it/MCS.R:72]: Hi @AldoGl, I committed a commented version of the MCS code

**Comment by AldoGl** [comment]
Thank you @mmartinoli87 for the new code, the comments are indeed very useful!

**Comment by AldoGl** [comment]
Thank you @muxator!

**Pull request merged**